exaudfclient/base/javacontainer/script_options/parser_legacy.cc

```diff
@@ -126,66 +131,64 @@ void ScriptOptionLinesParserLegacy::extractImportScripts(SwigFactory & swigFacto
     }
 }
 
-void ScriptOptionLinesParserLegacy::parseForScriptClass(std::function<void(const std::string &option)> callback,
-                                 std::function<void(const std::string&)> throwException) {
+void ScriptOptionLinesParserLegacy::parseForScriptClass(std::function<void(const std::string &option)> callback) {
+    try {
     parseForSingleOption(m_keywords.scriptClassKeyword(),
-                            [&](const std::string& value, size_t pos){callback(value);},
-                            [&](const std::string& msg){throwException("F-UDF-CL-SL-JAVA-1610" + msg);});
+                            [&](const std::string& value, size_t pos){callback(value);});
+    } catch (const std::runtime_error& ex) {
+        throw std::runtime_error(std::string("F-UDF-CL-SL-JAVA-1610") + ex.what());
+    }
 }
 
-void ScriptOptionLinesParserLegacy::parseForJvmOptions(std::function<void(const std::string &option)> callback,
-                                 std::function<void(const std::string&)> throwException) {
-   parseForMultipleOptions(m_keywords.jvmOptionKeyword(),
-                            [&](const std::string& value, size_t pos){callback(value);},
-                            [&](const std::string& msg){throwException("F-UDF-CL-SL-JAVA-1612" + msg);});
+void ScriptOptionLinesParserLegacy::parseForJvmOptions(std::function<void(const std::string &option)> callback) {
+    try {
+       parseForMultipleOptions(m_keywords.jvmOptionKeyword(),
+                                [&](const std::string& value, size_t pos){callback(value);});
+    } catch(const std::runtime_error& ex) {
+        throw std::runtime_error(std::string("F-UDF-CL-SL-JAVA-1612") + ex.what());
```

Comment:
Maybe a rethrow function might make sense

Reply:
Or crazy function that gets the code in the try block as lambda, returns the return value and catches exceptions and rethrows them with a additional error code

Reply:
Also thinking about if we should through a own exception like a OptionParserException

Reply:
Added a `rethrow` method.
-> The lambda maybe makes the code more difficult to read.

Reply:
Created class OptionParserException